gcc/d/gdc.texi

```diff
@@ -120,196 +120,662 @@ and the Info entries for @file{gdc} and @file{gcc}.
+@end table
 
-@item -fsplit-dynamic-arrays
-@cindex @option{-fsplit-dynamic-arrays}
-Split dynamic arrays into length and pointer when passing to functions.
+@item -fno-invariants
+@cindex @option{-finvariants}
+@cindex @option{-fno-invariants}
+Turns off code generation for class @code{invariant} contracts.
+
+@item -fno-moduleinfo
+@cindex @option{-fmoduleinfo}
+@cindex @option{-fno-moduleinfo}
+Turns off generation of the @code{ModuleInfo} and related functions
+that would become unreferenced without it, which may allow linking
+to programs not written in D.  Functions that will not be generated
+include module constructor and destructors (@code{static this} and
+@code{static ~this}), @code{unittest} code, and @code{DSO} registry
+functions for dynamically linked code.
 
-@item -femit-templates
-@cindex @option{-femit-templates}
-Control template emission behaviour.
+@item -fonly=@var{filename}
+@cindex @option{-fonly}
+Tells the compiler to parse and run semantic analysis on all modules
+on the command line, but only generate code for the module specified
+by @var{filename}.
 
-Turning on this option will emit all templates, but make them private
-to the translation unit. The executable will have multiple copies
-of code and data.
+@item -fno-postconditions
+@cindex @option{-fpostconditions}
+@cindex @option{-fno-postconditions}
+Turns off code generation for postcondition @code{out} contracts.
 
-Turning off this option will tell the compiler to not emit templates
-at all.
+@item -fno-preconditions
+@cindex @option{-fpreconditions}
+@cindex @option{-fno-preconditions}
+Turns off code generation for precondition @code{in} contracts.
 
-The default behaviour is to emit templates, but only if the compiler
-determines that it needs to.
+@item -frelease
+@cindex @option{-frelease}
+@cindex @option{-fno-release}
+Turns on compiling in release mode, which means not emitting runtime
+checks for contracts and asserts.  Array bounds checking is not done
+for @code{@@system} and @code{@@trusted} functions, and assertion
+failures are undefined behaviour.
+
+This is equivalent to compiling with following options:
+
+@gccoptlist{-fno-assert @gol
+-fbounds-check=safe @gol
+-fno-invariants @gol
+-fno-postconditions @gol
+-fno-preconditions @gol
+-fno-switch-errors @gol
+}
+
+@item -fno-switch-errors
+@cindex @option{-fswitch-errors}
+@cindex @option{-fno-switch-errors}
+This option controls what code should be generated when no case is
+matched in a @code{final switch} statement.  The default run time
+behavior is that a @code{SwitchError} will be thrown.  Turning off
+@option{-fswitch-errors} means that instead the execution of the
+program is immediately halted.
 
-@item -fdebug=@var{opt}
-@cindex @option{-fdebug}
-Compile in debug code into the program.
+@item -funittest
+@cindex @option{-funittest}
+@cindex @option{-fno-unittest}
+Turns on compilation of @code{unittest} code, and turns on the unittest
```

Comment:
Probably format: the `unittest` version identifier. Or somehow explicitly mention `version(unittest)`